### PR TITLE
Add documentation for query priority support by the transaction throttler

### PR DESCRIPTION
**Pull request by ejortegau**
This accompanies https://github.com/vitessio/vitess/pull/12662/

cc @harshit-gangal

### CODE REVIEW

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md [hunk 1]
```diff
@@ -119,3 +119,15 @@ As shown above, the metrics being instrumented this way are `vttablet_query_coun
...
 string. If `vttablet` is not started with `--enable-per-workload-table-metrics`, metrics are emitted without the
 workload label (e.g. `vttablet_query_counts{plan="Select",table="customer"}`.
+
+### Priority (`PRIORITY`)
+
+Specifies the priority associated with the execution of this query, as a number between 0 (lowest priority) and 100
+(highest priority). The priority is used by the transaction throttler (see `--tx-throttler-...` and `-enable-tx-throttler`
+flags in the `vttablet` documentation) to determine whether a particular query should be throttled. If the transaction
+throttler determines that a query may need to be throttled, it will throttle it with a probability of
+`1 - <query priority>`. This allows to avoid high business impact queries from being throttled (by setting their
```
Comment (harshit-gangal) [inline content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]:
```suggestion
`100 - <query priority>`. This allows avoiding high business impact queries from being throttled (by setting their
```

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md [hunk 2]
```diff
@@ -119,3 +119,15 @@ As shown above, the metrics being instrumented this way are `vttablet_query_coun
 If the query lacks the `WORKLOAD_NAME` directive, the corresponding label in the metric will have the value of an empty
 string. If `vttablet` is not started with `--enable-per-workload-table-metrics`, metrics are emitted without the
 workload label (e.g. `vttablet_query_counts{plan="Select",table="customer"}`.
+
+### Priority (`PRIORITY`)
+
+Specifies the priority associated with the execution of this query, as a number between 0 (lowest priority) and 100
+(highest priority). The priority is used by the transaction throttler (see `--tx-throttler-...` and `-enable-tx-throttler`
+flags in the `vttablet` documentation) to determine whether a particular query should be throttled. If the transaction
```
Comment (harshit-gangal) [inline content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]: let's link the vttablet documentation here

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md (hunk 1 above)
Comment (ejortegau) [reply content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]: Fixed

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md (hunk 2 above)
Comment (ejortegau) [reply content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]: Fixed.

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md [hunk 3]
```diff
@@ -119,3 +119,15 @@ As shown above, the metrics being instrumented this way are `vttablet_query_coun
...
 string. If `vttablet` is not started with `--enable-per-workload-table-metrics`, metrics are emitted without the
 workload label (e.g. `vttablet_query_counts{plan="Select",table="customer"}`.
+
+### Priority (`PRIORITY`)
+
+Specifies the priority associated with the execution of this query, as a number between 0 (lowest priority) and 100
+(highest priority). The priority is used by the transaction throttler (see `--tx-throttler-...` and `-enable-tx-throttler`
+flags in the `vttablet` [documentation](../../reference/programs/vttablet.md))) to determine whether a particular query
+should be throttled. If the transaction throttler determines that a query may need to be throttled, it will throttle it
+with a probability of `1 - <query priority>`. This allows avoiding high business impact queries from being throttled (by
```
Comment (harshit-gangal) [inline content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]: I believe this is still incorrect ``1 - <query priority>``

Comment (ejortegau) [reply content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]: Right, I missed this. Fixed now.

**Review by harshit-gangal — APPROVED** [review]

File: content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md [hunk 4]
```diff
@@ -119,3 +119,15 @@ As shown above, the metrics being instrumented this way are `vttablet_query_coun
 If the query lacks the `WORKLOAD_NAME` directive, the corresponding label in the metric will have the value of an empty
 string. If `vttablet` is not started with `--enable-per-workload-table-metrics`, metrics are emitted without the
 workload label (e.g. `vttablet_query_counts{plan="Select",table="customer"}`.
+
+### Priority (`PRIORITY`)
+
+Specifies the priority associated with the execution of this query, as a number between 0 (lowest priority) and 100
+(highest priority). The priority is used by the transaction throttler (see `--tx-throttler-...` and `-enable-tx-throttler`
```
Comment (harshit-gangal) [inline content/en/docs/17.0/user-guides/configuration-advanced/comment-directives.md]:
```suggestion
Specifies the priority associated with the execution of this query, as a number between 0 (highest priority) and 100
(lowest priority). The priority is used by the transaction throttler (see `--tx-throttler-...` and `-enable-tx-throttler`
```

File: content/en/docs/17.0/reference/programs/vttablet.md [hunk 5]
```diff
@@ -401,6 +401,7 @@ The following global options apply to `vttablet`:
...
 | --tx-throttler-config                                            | string        | Synonym to -tx_throttler_config (default "target_replication_lag_sec: 2\nmax_replication_lag_sec: 10\ninitial_rate: 100\nmax_increase: 1\nemergency_decrease: 0.5\nmin_duration_between_increases_sec: 40\nmax_duration_between_increases_sec: 62\nmin_duration_between_decreases_sec: 20\nspread_backlog_across_sec: 20\nage_bad_rate_after_sec: 180\nbad_rate_increase: 0.1\nmax_rate_approach_threshold: 0.9\n")                                                                                        |
+| --tx-throttler-default-priority int                              | int           | Default query priority used by the transaction throttler if it is not specified in a query comment directive. It must be a number between 0 (lowest priority) and 100 (highest priority) (default: 0)                                                                                                                                                                                                                                                                                                      |
```
Comment (harshit-gangal) [inline content/en/docs/17.0/reference/programs/vttablet.md]:
```suggestion
| --tx-throttler-default-priority int                              | int           | Default query priority used by the transaction throttler if it is not specified in a query comment directive. It must be a number between 0 (highest priority) and 100 (lowest priority) (default: 100)                                                                                                                                                                                                                                                                                                      |
```

**Review by harshit-gangal — APPROVED** [review]

**Comment by shlomi-noach** [comment]
Hmmm, I can't seem to be able to restart CI. Also, I pushed an empty commit to [eduardo.ortega/txthrottler_query_priority](https://github.com/ejortegau/vitess-website/tree/eduardo.ortega/txthrottler_query_priority) branch on [ejortegau/vitess-website](https://github.com/ejortegau/vitess-website) but it does not show up.

@ejortegau can you please try and kick this branch? Push an empty commit or something?

**Comment by ejortegau** [comment]
Pushed, waiting to see if CI runs.

**Comment by ejortegau** [comment]
It seems that CI run is stuck :-(

**Pull request merged**